app/main/controller/presto_controller.py

```diff
@@ -26,8 +26,9 @@ def post(self, action, model_type):
         data = request.args or request.json
         app.logger.info(f"PrestoResource {action}")
         if action == "add_item":
-            result = similarity.callback_add_item(data, model_type)
-            Webhook.return_webhook(app.config['CHECK_API_HOST'], action, model_type, result)
+            result = similarity.callback_add_item(dict(**data.get("body"), **data.get("response", {})), model_type)
+            if data.get("body", {}).get("requires_callback"):
+                Webhook.return_webhook(app.config['CHECK_API_HOST'], action, model_type, result)
```

Comment:
instead of `requires_callback`, maybe this could just be `callback`  and the caller could set the callback url?  i.e. check would set it as CHECK_API_HOST and timpani would set it to match https://github.com/meedan/timpani/blob/6134e47866fb72a58ecdc0ce36977425abffd728/timpani/conductor/app.py#L111

Reply:
I'm not sure about that - I think that conceptually disentangling it may end up being useful? I can see cases in my head where we don't want to / wont have the flexibility of deleting the callback URL from the other side of the request, but also want to be able to deny the callback anyways? I think they are two distinct ideas. I am happy to change that if everyone disagrees.